### PR TITLE
Add support for const values in StringMap

### PR DESCRIPTION
**Pull request by nordlow**
For sake of compliance with `T[string]` when `is(T == const)`.

The primary need is `const(T)` where `is(T == class)`.

Ready for prel review, @9il.

### CODE REVIEW

File: source/mir/string_map.d [hunk 1]
```diff
@@ -632,7 +632,7 @@ struct StringMap(T, U = uint)
                     assert (index < length);
                     index = implementation._indices[index];
                     assert (index < length);
-                    implementation._values[index] = move(value);
+					move(cast()value, cast()(implementation._values[index]));
```
Comment (9il) [inline source/mir/string_map.d:635]: use spaces please

Comment (nordlow) [reply source/mir/string_map.d:635]: Sorry about that. Gonna adjust.

Comment (nordlow) [reply source/mir/string_map.d:635]: Adjusted now.

**Comment by nordlow** [comment]
In essence it's safe to cast away const on a parameter passed by value.

**Pull request merged**